kafka-ui-react-app/src/components/Schemas/Edit/__tests__/Edit.spec.tsx

```diff
@@ -0,0 +1,103 @@
+import { shallow } from 'enzyme';
+import { SchemaType } from 'generated-sources';
+import React from 'react';
+import Edit from '../Edit';
+
+describe('Edit Component', () => {
+  const mockSchema = {
+    subject: 'Subject',
+    version: '1',
+    id: 1,
+    schema: '{"schema": "schema"}',
+    compatibilityLevel: 'BACKWARD',
+    schemaType: SchemaType.AVRO,
+  };
+
+  describe('when schemas are not fetched', () => {
+    const component = shallow(
+      <Edit
+        subject="Subject"
+        clusterName="ClusterName"
+        schemasAreFetched={false}
+        fetchSchemasByClusterName={jest.fn()}
+        updateSchemaCompatibilityLevel={jest.fn()}
+        createSchema={jest.fn()}
+        schema={mockSchema}
+      />
+    );
+    it('matches the snapshot', () => {
+      expect(component).toMatchSnapshot();
+    });
+    it('shows loader', () => {
+      expect(component.find('PageLoader').exists()).toBeTruthy();
+    });
+  });
+
+  describe('when schemas are fetched', () => {
+    const component = shallow(
+      <Edit
+        subject="Subject"
+        clusterName="ClusterName"
+        schemasAreFetched
+        fetchSchemasByClusterName={jest.fn()}
+        updateSchemaCompatibilityLevel={jest.fn()}
+        createSchema={jest.fn()}
+        schema={mockSchema}
+      />
+    );
+    it('matches the snapshot', () => {
+      expect(component).toMatchSnapshot();
+    });
+    it('shows editor', () => {
+      expect(component.find('JSONEditor').length).toEqual(2);
+      expect(component.find('button').exists()).toBeTruthy();
+    });
+    it('does not call createSchema on button click without changing the schema', () => {
+      const mockCreateSchema = jest.fn();
+      const componentWithMockFn = shallow(
```

Comment:
To check whether the effect is called or not we need to use mount instead of shallow